### PR TITLE
[Release, ROCm] release ROCm docker build for AMD MI GPUs

### PR DESCRIPTION
**Pull request by HaiShaw**
## Motivation

Add ROCm docker build to release process, and push to dockerhub repo: `lmsysorg/sglang`

## Modifications

Introduce new `release-docker-amd.yml` using runner `docker-build-amd`.

## Checklist

- [+] Format your code according to the [Contributor Guide](https://github.com/sgl-project/sglang/blob/main/docs/references/contributor_guide.md).
- [+] Add unit tests as outlined in the [Contributor Guide](https://github.com/sgl-project/sglang/blob/main/docs/references/contributor_guide.md).
- [+] Update documentation as needed, including docstrings or example tutorials.

### CODE REVIEW

**Comment by HaiShaw** [comment]
@merrymercy thanks for review!

File: .github/workflows/release-docker-amd.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
+name: Release Docker Images
```
Comment (merrymercy) [inline .github/workflows/release-docker-amd.yml]:
```suggestion
name: Release Docker Images (AMD)
```

File: .github/workflows/release-docker-amd.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    branches:
+      - main
+    paths:
+      - "python/sglang/version.py"
+  workflow_dispatch:
+
+jobs:
+  publish:
+    if: github.repository == 'sgl-project/sglang'
+    runs-on: docker-build-amd
```
Comment (merrymercy) [inline .github/workflows/release-docker-amd.yml]:
```suggestion
    runs-on: docker-builder-amd
```

**Review by merrymercy — CHANGES_REQUESTED** [review]

**Comment by merrymercy** [comment]
@zhyncs Can you help test this?

**Comment by merrymercy** [comment]
Actually, let me merge this first so we can start debugging

**Comment by merrymercy** [comment]
Running at https://github.com/sgl-project/sglang/actions/runs/11738482561

**Comment by merrymercy** [comment]
@HaiShaw The build is successful. Can you try to pull the images from lmsys?

**Comment by HaiShaw** [comment]
@merrymercy , image runnable great on MI300X. Thanks!

**Comment by zhyncs** [comment]
Nice work!

**Pull request merged**